From 1f845d211b8406a08ede5b45b91761e78554c6e1 Mon Sep 17 00:00:00 2001
From: Robert Nelson <robertcnelson@gmail.com>
Date: Tue, 8 Mar 2016 11:55:34 -0600
Subject: [PATCH 4/4] net/wireless: SanCloud wifi: issue when associating with
 an SSID when there is also a hidden SSID in the scan list

Signed-off-by: Robert Nelson <robertcnelson@gmail.com>
---
 net/wireless/scan.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/net/wireless/scan.c b/net/wireless/scan.c
index 14d5369..324b2cc 100644
--- a/net/wireless/scan.c
+++ b/net/wireless/scan.c
@@ -404,7 +404,7 @@ static bool is_bss(struct cfg80211_bss *a, const u8 *bssid,
 	if (bssid && !ether_addr_equal(a->bssid, bssid))
 		return false;
 
-	if (!ssid)
+	if ( (!ssid) || (!ssid_len) )
 		return true;
 
 	ies = rcu_access_pointer(a->ies);
-- 
2.7.0

